suites/apis/submitAndQueryNodesTest.js

```diff
@@ -193,6 +193,44 @@ Scenario('test with_path_to - last to first node @reqData', async (peregrine, sh
 }).retry(2);
 
 
+Scenario('submit data node with consent codes @indexRecordConsentCodes', async (sheepdog, indexd, nodes) => {
+  metadata = nodes.getFileNode().clone();
+  if (metadata.data.consent_codes) {
+      metadata.data.consent_codes = ["CC1", "CC2"];
+  }
+
+  // assuming all data files can be submitted with a single link to a
+  // core_metadata_collection: add it, and remove other links
+  const cmcSubmitterID = await nodes.generateAndAddCoremetadataNode(sheepdog);
+  console.log("cmcSubmitterID: " + cmcSubmitterID);
+  for (var prop in metadata.data) {
+    if (metadata.data.hasOwnProperty(prop) && metadata.data[prop].hasOwnProperty('submitter_id')) {
+      console.log("Deleting: " + metadata.data[prop]);
+      delete metadata.data[prop];
+    }
+  }
+  metadata.data.core_metadata_collections = {
+    submitter_id: cmcSubmitterID
+  };
+
+  await sheepdog.complete.addNode(metadata);
```

Comment:
just because I would need to pass in a (fake) `fileGuid`, which I think would change the flow in sheepdog (update instead of create)? Either way sheepdog will create a record if one doesn't exist, so it ends up being the same end result, but only because sheepdog doesn't find that guid

Reply:
mm i see. what if you pass it `null`? and (if needed) update `submitGraphAndFileMetadata` to not add `object_id` if `!fileGuid`
sorry i'm insisting on this, it's just a lot of code duplication

Reply:
(also fake file size, hash etc)

Reply:
i still think you could update `submitGraphAndFileMetadata` to have `null` default values and not update `metadata` did/size/hash if the values are null... imo it would make the function more general and useful

or if you don't like that you could pass in `nodes.getFileNode().data`'s size and hash